src/pages/en/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -2,67 +2,576 @@
+
+
+### Put your code in `src`
+
+1. **Delete** Gatsby's `public/` folder. 
+    
+    Gatsby uses the `public/` directory for its build output, so you can safely discard this folder. You will no longer need a built version of your Gatsby site. (Astro uses `dist/` by default for the build output.)
+
+2. **Rename** Gatsby's `static/` folder to `public/`, and use it as Astro's `public/` folder. 
+
+    Astro uses a folder called `public/` for static assets. You can alternatively copy the contents of `static/` into your existing Astro `public/` folder.
+
+3. **Copy or Move** Gatsby's other files and folders (e.g. `components`, `pages` etc.) as needed into your Astro `src/` folder as you rebuild your site, following [Astro's project structure](/en/core-concepts/project-structure/).
+
+  Astro's `src/pages/` folder is a special folder used for file-based routing to create your site's pages and posts from `.astro`, `.md` and `.mdx` files. You will not have to configure any routing behavior for your Astro, Markdown and MDX files.
+
+  All other folders are optional, and you can organize the contents of your `src/` folder any way you like. Other common folders in Astro projects include `src/layouts/`, `src/components`, `src/styles`, `src/scripts`.
+
+
+### Convert JSX files to `.astro` files
+
+Here are some tips for converting a Gatsby `.js` component into a `.astro` component:
+
+1. Use only the `return()` of the existing Gatsby component function as your HTML template.
+
+1. Change any [Gatsby or JSX syntax to Astro](#reference-convert-to-astro-syntax) (e.g. `<Link to="">`, `{children}`, `className`, inline style objects) or to HTML web standards.
+
+1. Move any necessary JavaScript, including import statements, into a "code fence" (`---`). Note: JavaScript to [conditionally render content](/en/core-concepts/astro-components/#dynamic-html) is often written inside the HTML template directly in Astro.
+
+4. Write an `Astro.props` statement to access any additional props that were previously passed to your Gatsby function.
+
+4. Decide whether any imported components also need to be converted to Astro. With the official integration installed, you can [use existing React components in your Astro file](/en/guides/integrations-guide/react/). But, you may want to convert them to Astro, especially if they do not need to be interactive!
+
+4. Remove any GraphQL queries. Instead, use import and `Astro.glob()` statements to query your local files. Update any [dynamic HTML content variables](/en/core-concepts/astro-components/#dynamic-html) to use Astro-specific properties instead.
+
+See [an example from Gatsby's Blog starter template converted step-by-step](#guided-example-gatsby-layout-to-astro)
+
+#### Compare: `.jsx` vs `.astro`
+
+Compare the following Gatsby component and a corresponding Astro component:
+
+<AstroJSXTabs>
```

Comment:
Not convinced about splitting these into tabs making it _impossible_ to see both at the same time. (OK, I guess you could open the page twice in two windows? 😅)

Two notes:

1. How about we break this up a little and show JSX followed by Astro for each related “chunk”? Although there is some advantage obviously to seeing the full component in one block.

2. I think we can def simplify this comparison. Quick wins, but we can probably do more:
    - just one component import instead of header _and_ footer
    - _maybe_ save GraphQL for later?
    - only use one `style={{}}` object, and make it only include one/two styles

Perhaps 2 is enough here and we condense it down, show them consecutively and call it a day.

Reply:
I continue to struggle here. I *like* an entire file comparison. I *like* that it shows GraphQL (because that's a big deal!) compared to `Astro.glob()`. I even sort of like that you flip the tab and realize that you CAN see all the `.astro` file on one desktop screen at once, but you can't with the original?  :sweat: 

I wish we had a side-by-side strategy, but we don't. At this point, you can't see both on the page at the same time anyway (you'd have to scroll, they're too long, and will be even with some cutting). SO, at least flipping tabs is a nicer way to compare than scrolling, I actually think? If the point is just looking as a comparison, there's something to be said for the page not moving on you?

I'll think about where it could be cut to be shorter, but given that it won't get THAT much shorter, I'm not even sure that's our biggest problem.

We'll figure this one out eventually!

Reply:
I agree with @delucis's #2. The tabs interface is fine, but it works better when you don't have to scroll. We can simplify the example to show "one-of-a-kind" - we don't need two style objects each with four properties, lengthening the sample without adding a new concept.

Reply:
Will trim this one!